query-engine/driver-adapters/executor/src/recording.ts

```diff
@@ -7,6 +7,20 @@ import {
 
 type Recordings = ReturnType<typeof createInMemoryRecordings>;
 
+/* 
+  This is a workaround for the fact that BigInts are not serializable by JSON.stringify.
+  We need to add a toJSON method to BigInt so that it can be serialized as part of serializing a query.
+ */
+declare global {
+  interface BigInt {
+    toJSON(): number;
+  }
+}
+
+BigInt.prototype.toJSON = function () {
```

Comment:
Rather than patching up prototype, you can add a second argument to `JSON.stringify`:

```ts
JSON.stringify(data, (value) => {
   if (typeof value === 'bigint') {
       return String(value)
   }
   return value
})
```

Reply:
I know, 

I understand the disadvantage of monkeypatching is to not let patched types behave unexpectedly in different parts of the codebase (before / after patch), but given that BigInt is not  _stringify-able_, and we would need to repeat that wherever we are serializing bigints, what would be the advantage of making the change in the callsites as you suggest?

Reply:
It would at least make it clear that we are doing something different for bigint than what normal `JSON.stringify` does and it makes it easy to see what exectly is different. With monkey-patch, this information is hidden somewhere in the random file where monkey-patching happens.

Reply:
Also, going through `Number` rather than string will lose data on > 53 Bit. Might not be relevant for benchmarks now, but could make for a confusing surprise in the future.

Reply:
Yes, I'm aware of that, this for benchmarks is all acceptable and good enough. 

Nevertheless thanks for clarifying, JSON.stringify skews benchmarks by 2%, so I'm finding another approxination. Moving this to draft.